File: python/zig_zag.py
import time
#working...........
#success..............
def delay():#this function is working...
#it considers milliseconds...
	time.sleep(0.05)
def style_l_to_r():#left to right is working..
	for x in range(0,15):
		print("   "*x, end="#\n")
		delay()
def style_r_to_l():#right to left is working..
	for k in reversed(range(15)):
		print("   "*k, end="#\n")
		delay()
#now the main body...
def main():
	tr=True
	while tr==True:
		style_l_to_r()
		style_r_to_l()
main()

File: java/zig_zag.java
 /*
 working.........
 success.........*/
 class style extends Thread{
	public static void main(String h[]){
		style s=new style();
		s.start();
	}
	public void run(){
		try{
			for(;;){
		for(int i=0;i<15;i++){//this one is working...
			for(int f=0;f<i;f++){
				System.out.print("  ");
			}
			System.out.println("#");
			Thread.sleep(60);
		}
		for(int e=15;e>0;e--){
			for (int q=0;q<e ;q++ ) {
				System.out.print("  ");
			}
			System.out.println("#");
			Thread.sleep(60);
		}
	}
}
	catch(InterruptedException e){System.out.println(e);}
	}
}
public class zig_zag{
	public static void main(String k[]){
		style s=new style();
		s.start();
	}
}